### PR TITLE
Web Inspector: Dark Mode: Increase the contrast in elements tab for @media (prefers-contrast: more) and @media (prefers-color-scheme: dark)

### PR DESCRIPTION
**Pull request by francescorn**
#### 4b04e9d474e7e62186e7c89afbc2def28deba1b0
<pre>
Web Inspector: Dark Mode: Increase the contrast in elements tab for @media (prefers-contrast: more) and @media (prefers-color-scheme: dark)
<a href="https://bugs.webkit.org/show_bug.cgi?id=271722">https://bugs.webkit.org/show_bug.cgi?id=271722</a>

Reviewed by NOBODY (OOPS!).

Lighten box model borders when hovered in BoxModelDetailsSectionRow.css.
Lighten borders in the box model in BoxModelDetailsSectionRow.css.
Darken panels in ChangesDetailsSidebarPanel.css.
Create --box-model-border-color in BoxModelDetailsSectionRow.css.
Create --box-model-border-color-hovered in BoxModelDetailsSectionRow.css.
Replace --has-border-radius-has-border-radius with --box-model-border-radius in BoxModelDetailsSectionRow.css.
Lighten selector to hsl(0, 0%, 90%) in ComputedStyleSection.css.
Lighten pseudo elements in DomTreeOutline.css.
Darken shadow in DomTreeOutline.css.
Lighten range value in FontVariationDetailsSectionRow.css.
Lighten background color in GeneralStyleDetailsSidebarPanel.css.
Lighten background color of options-container when selected and hovered, in GeneralStyleDetailsSidebarPanel.css.
Increase the brightness of quick console and the showing log in QuickConsole.css.

* Source/WebInspectorUI/UserInterface/Views/BoxModelDetailsSectionRow.css:
(.details-section .row.box-model .box.border):
(.details-section .row.box-model .box.border.has-top-left-radius,):
(.details-section .row.box-model .box.border.has-top-right-radius,):
(.details-section .row.box-model .box.border.has-bottom-right-radius,):
(.details-section .row.box-model .box.border.has-bottom-left-radius,):
(@media (prefers-color-scheme: dark) .details-section .row.box-model .box.border):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model .box.margin):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model .box.border):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model .box):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model:not(.hovered) .box.margin,):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model:not(.hovered) .box.padding,):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model:not(.hovered) .box.border,):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model:is(.hovered) .box.margin,):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model:is(.hovered) .box.padding,):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model:is(.hovered) .box.border,):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model:is(.hovered) .box.position,):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .details-section .row.box-model:is(.hovered) .box.content,):
* Source/WebInspectorUI/UserInterface/Views/ChangesDetailsSidebarPanel.css:
(@media (prefers-contrast: more) @media (prefers-color-scheme: dark) .changes-panel .css-property-line.added):
* Source/WebInspectorUI/UserInterface/Views/ComputedStyleSection.css:
(@media (prefers-contrast: more) @media (prefers-color-scheme: dark) .computed-style-section .computed-property-item .property-trace-item .selector):
* Source/WebInspectorUI/UserInterface/Views/DOMTreeOutline.css:
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .tree-outline.dom):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .tree-outline.dom li.inspected-node &gt; span::after):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) body:not(.window-inactive, .window-docked-inactive) .tree-outline.dom:focus-within li.inspected-node.selected &gt; span::after):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .tree-outline.dom .shadow):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .tree-outline.dom .html-pseudo-element):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .showing-find-banner .tree-outline.dom .search-highlight):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .tree-outline.dom li .pseudo-class-indicator):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .content-view.dom-tree.deemphasize-unrendered .tree-outline.dom li:not(.rendered, .selected) &gt; span &gt; *:not(.html-comment), .content-view.dom-tree.deemphasize-unrendered .tree-outline.dom:not(:focus-within) li.selected:not(.rendered) &gt; span &gt; *:not(.html-comment), body:is(.window-inactive, .window-docked-inactive) .content-view.dom-tree.deemphasize-unrendered .tree-outline.dom li.selected:not(.rendered) &gt; span &gt; *:not(.html-comment)):
* Source/WebInspectorUI/UserInterface/Views/FontVariationDetailsSectionRow.css:
(@media (prefers-contrast: more) @media (prefers-color-scheme: dark) .details-section &gt; .content &gt; .group &gt; .row.font-variation &gt; .variation-range &gt; input[type=&quot;range&quot;]):
* Source/WebInspectorUI/UserInterface/Views/GeneralStyleDetailsSidebarPanel.css:
(.multi-sidebar.showing-multiple &gt; .sidebar &gt; .panel.details:not(.style-rules) &gt; .options-container &gt; .toggle,):
(@media (prefers-color-scheme: dark) .sidebar &gt; .panel.details.css-style &gt; .content ~ .options-container &gt; .new-rule):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .sidebar &gt; .panel.details.css-style &gt; .content ~ .options-container &gt; .toggle:is(.selected, :hover)):
(@media (prefers-color-scheme: dark) @media (prefers-contrast: more) .sidebar &gt; .panel.details.css-style &gt; .content ~ .options-container &gt; .toggle:is(.selected, :hover)::before):
</pre>
https://github.com/WebKit/WebKit/commit/4b04e9d474e7e62186e7c89afbc2def28deba1b0

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/47901 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27102 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/50744 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/50582 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/43954 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/32973 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/24587 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/38965 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48483 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/24761 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/41415 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20265 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22239 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/42596 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/5948 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/44248 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/43030 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52476 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/22943 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/19286 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/46273 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24213 "Built successfully") | | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45316 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/25003 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/23934 "Built successfully") | | |

### CODE REVIEW

File: Source/WebInspectorUI/UserInterface/Views/BoxModelDetailsSectionRow.css [hunk 1]
```diff
@@ -188,5 +189,58 @@
...
+        .details-section .row.box-model .box.position.active {
+            border-color: white;
+        }
+
+        .details-section .row.box-model:is(.hovered) .box.content,
+        .details-section .row.box-model .box.content.active {
+            border-color: white;
+        }
+    }
+}
```
Comment (rcaliman-apple) [inline Source/WebInspectorUI/UserInterface/Views/BoxModelDetailsSectionRow.css]: Code duplication like this only to set different colors makes the file larger and more difficult to read through.

Could we define some CSS variables for with color values and switch between their values in a `@media` block? Similar to what you did in `Variables.css`

File: Source/WebInspectorUI/UserInterface/Views/Variables.css [hunk 2]
```diff
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2015-2016 Apple Inc. All rights reserved.
+ * Copyright (C) Frances Cornwall.
```
Comment (rcaliman-apple) [inline Source/WebInspectorUI/UserInterface/Views/Variables.css]: If you want to add yourself to the Copyright header, this should also include a year.

**Review by rcaliman-apple — COMMENTED** [review]
Thanks for taking this on! Some notes and questions before a more thorough review:

macOS supports a high contrast mode where colors are changed system-wide. Web Inspector tries to follow the system values wherever possible to ensure a cohesive UI. Are the color values in this patch matching the ones from the system?

Screenshot 2024-03-28 at 17 27 36


Web Inspector has some custom UI elements where it's not immediately clear which system color should apply. As a rule of thumb, following Xcode's colors for similar UI (for example, breakpoint colors) is a good proxy. 

I see there are three patches addressing this change across various parts of Web Inspector. Would you like to combine them into a single one?

File: Source/WebInspectorUI/UserInterface/Views/BoxModelDetailsSectionRow.css (hunk 1 above)
Comment (francescorn) [reply Source/WebInspectorUI/UserInterface/Views/BoxModelDetailsSectionRow.css]: I replaced them with two CSS variables: `--box-model-border-color` and `--box-model-border-color-hovered` @rcaliman-apple.

**Comment by francescorn** [comment]
@rcaliman-apple The color values in this patch match the ones in the system in dark mode, the prs address increased contrast for dark mode.

I am splitting [pr](https://github.com/WebKit/WebKit/pull/26374) into multiple prs for easier reviewing.

File: Source/WebInspectorUI/UserInterface/Views/BoxModelDetailsSectionRow.css (hunk 1 above)
Comment (francescorn) [reply Source/WebInspectorUI/UserInterface/Views/BoxModelDetailsSectionRow.css]: Are the CSS variables okay? @rcaliman-apple

File: Source/WebInspectorUI/UserInterface/Views/Variables.css (hunk 2 above)
Comment (francescorn) [reply Source/WebInspectorUI/UserInterface/Views/Variables.css]: Thank you for letting me know, I added 2024 in all of my copyrights @rcaliman-apple.

**Comment by francescorn** [comment]
Here are my changes in the mini-browser on the left and Safari 17.4.1 on the right. I uploaded it to YouTube, the file size is too large on GitHub, even at 480p: https://youtu.be/4IWnggoXQX0

**Pull request closed without merging**